app/scripts/modules/kubernetes/src/v2/instance/details/details.controller.ts

```diff
@@ -0,0 +1,112 @@
+import { IController, IPromise, IQService, module } from 'angular';
+import { IModalService } from 'angular-ui-bootstrap';
+import { flattenDeep } from 'lodash';
+
+import {
+  Application,
+  CONFIRMATION_MODAL_SERVICE,
+  INSTANCE_READ_SERVICE,
+  InstanceReader,
+  RECENT_HISTORY_SERVICE,
+  RecentHistoryService
+} from '@spinnaker/core';
+
+import { IKubernetesInstance } from './IKubernetesInstance';
+
+interface InstanceFromStateParams {
+  instanceId: string;
+}
+
+interface InstanceManager {
```

Comment:
This is a better name than what I had for App Engine. If you wanted to change it there...that would be OK.